### ISSUE TITLE
Malformed ESSIDs and APs with channel 108 in airodump-ng

### ISSUE THREAD

**Issue by aircrack-ng** [aircrack-ng/aircrack-ng#1244]
**Reported by aifjdaoifjsdoif on 24 Sep 2013 22:29 UTC**

Here's my current setup:

 * Arch Linux x86_64, Linux 3.11.1
 * Alfa AWUS036NHA (Atheros AR9271, ID 0cf3:9271)
 * Aircrack-ng from both SVN and 1.1

I start my card in monitor mode:

```
$ sudo systemctl stop NetworkManager
$ sudo killall wpa_supplicant
$ sudo airmon-ng start wlan0

Interface   Chipset     Driver

wlan0       Atheros     ath9k - [               (monitor mode enabled on mon0)
```

And run `airodump-ng`:

```
$ sudo airodump-ng mon0

 CH  1 ](phy2]
)[ Elapsed: 4 s ][ 2013-09-24 14:01

 BSSID              PWR  Beacons    #Data, #/s  CH  MB   ENC  CIPHER AUTH ESSID

 XX:XX:XX:XX:XX:XX  -85        0        2    0 108  -1   WPA              <length:  0>
 XX:XX:XX:XX:XX:XX  -87        3        0    0   9  54 . WPA2 CCMP   PSK  95.&<7E......{>-...UiQ-
 XX:XX:XX:XX:XX:XX  -77        8        2    0  11  54e  WPA2 CCMP   PSK  2WIR123
 ...
```

Most of the access points are okay, but all the other ones show up either like the first or the second one. `2WIR123` will sometimes turn into `2WIRE123` after a few seconds, then back into `2I123`. The network without an ESSID will just stay there for a while. Sometimes a client pops up that appears to be connected to it.

I've had this problem with two wireless cards now, so unless I've been having really good luck, I suspect this is a problem with aircrack.

Let me know if you need any more info.

**Comment by aircrack-ng** [comment]
**Comment by anonymous on 24 Sep 2013 22:29 UTC**

I also had the problem with really strange ESSIDs once. At the same time, my wireless card crashed so I put it on a bug there. As I was recently looking around in airodump code, everything is a big mess so no wonder there might be some mistakes in SSID handling. You may want to have a look at the SSID parsing in the big dump_add_packet function, there might be something wrong with some lengths and there are definitively missing checks.

Channel problem fixed in ticket #670 (see my last posts).

**Comment by aircrack-ng** [comment]
**Comment by misterx on 24 Sep 2013 22:29 UTC**

Fixed in aircrack-ng/aircrack-ng@5f23bdad0e9d640ae87ca35edb20d3ba90ace81b.
